vnext/Microsoft.ReactNative/Utils/BatchingEventEmitter.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "JSValue.h"
+#include "ReactHost/React.h"
+#include "ReactPropertyBag.h"
+#include "winrt/Microsoft.ReactNative.h"
+
+#include <deque>
+#include <mutex>
+
+namespace winrt::Microsoft::ReactNative::implementation {
+struct BatchedEvent {
+  int64_t tag{};
+  winrt::hstring eventName;
+  JSValueObject eventObject;
+};
+} // namespace winrt::Microsoft::ReactNative::implementation
+
+namespace winrt::Microsoft::ReactNative {
+
+//! Emits queued events from native to JS in queued batches (at most once per-native frame). Events within a batch may
+//! be coalesced. The batch is finished at the time the JS thread starts to process it. I.e. it is possible for a batch
+//! to last for multiple frames if the JS thread is blocked. This is by-design as it allows our coalescing strategy to
+//! account for long operations on the JS thread.
+struct BatchingEventEmitter : winrt::implements<BatchingEventEmitter, winrt::Windows::Foundation::IInspectable> {
```

Comment:
>winrt::implements [](start = 30, length = 17)

I wonder why it needs to be a winrt class. It does not implement any WinRT interfaces. Why not to have a simple class and then use std::make_unique or std::make_shared to instantiate it.

Reply:
wouldn't we want this facility to be available to apps / modules written in C#?

Reply:
The first iteration exposed this to ABI. ACoates pointed out that keeping this as an internal API lets us iterate on it though, so we do not expose it yet. We should consider doing this if we see customers wanting to do this though.